File: ts_sdk/src/claude_hook_events/type-guards.ts
/**
 * Type guard functions for HookEventData.
 */

import type { HookEventData } from './hook-event-data';
import { HookEventType } from './event-types';

/** Check if event is a tool event (has tool_name). */
export function isToolEvent(h: HookEventData): boolean {
  return !!h.tool_name;
}

/** Check if event is PreToolUse. */
export function isPreToolUse(h: HookEventData): boolean {
  return h.hook_event_name === HookEventType.PRE_TOOL_USE;
}

/** Check if event is PostToolUse. */
export function isPostToolUse(h: HookEventData): boolean {
  return h.hook_event_name === HookEventType.POST_TOOL_USE;
}

/** Check if event is Notification. */
export function isNotification(h: HookEventData): boolean {
  return h.hook_event_name === HookEventType.NOTIFICATION;
}

/** Check if event is Stop. */
export function isStop(h: HookEventData): boolean {
  return h.hook_event_name === HookEventType.STOP;
}

/** Check if event is UserPromptSubmit. */
export function isUserPromptSubmit(h: HookEventData): boolean {
  return h.hook_event_name === HookEventType.USER_PROMPT_SUBMIT;
}

/** Check if event is SubagentStart or SubagentStop. */
export function isSubagentEvent(h: HookEventData): boolean {
  return (
    h.hook_event_name === HookEventType.SUBAGENT_START ||
    h.hook_event_name === HookEventType.SUBAGENT_STOP
  );
}

File: flow_sdk/claude_hook_events/type_guards.py
"""Type guard functions for HookEventData."""

from __future__ import annotations

from flow_sdk.claude_hook_events.hook_event_data import HookEventData


def is_tool_event(h: HookEventData) -> bool:
    """Check if event has a tool_name (tool event)."""
    return h.tool_name is not None


def is_pre_tool_use(h: HookEventData) -> bool:
    """Check if event is PreToolUse."""
    return h.hook_event_name == "PreToolUse"


def is_post_tool_use(h: HookEventData) -> bool:
    """Check if event is PostToolUse."""
    return h.hook_event_name == "PostToolUse"


def is_notification(h: HookEventData) -> bool:
    """Check if event is Notification."""
    return h.hook_event_name == "Notification"


def is_stop(h: HookEventData) -> bool:
    """Check if event is Stop."""
    return h.hook_event_name == "Stop"


def is_user_prompt_submit(h: HookEventData) -> bool:
    """Check if event is UserPromptSubmit."""
    return h.hook_event_name == "UserPromptSubmit"


def is_subagent_stop(h: HookEventData) -> bool:
    """Check if event is SubagentStop."""
    return h.hook_event_name == "SubagentStop"


def is_session_start(h: HookEventData) -> bool:
    """Check if event is SessionStart."""
    return h.hook_event_name == "SessionStart"


def is_session_end(h: HookEventData) -> bool:
    """Check if event is SessionEnd."""
    return h.hook_event_name == "SessionEnd"
